### PR TITLE
Fix: modern - slider:fix possible wrong responsive images URL in mult…

### LINKED ISSUE #1247: Multisite: possible wrong responsive images URL in main sliders
Reported here:
https://secure.helpscout.net/conversation/451954896/33519?folderId=607709

Might not be visible at the moment on the user's website, but analyzing the source we can see some URLs in the srcset are wrong:
`https://educationforasia.org/wp-content/uploads/sites/12/2016/07/education-in-asia.jpg 1170w, https://educationforasia.org/wp-content/uploads/sites/12/2016/07/education-in-asia-300x128.jpg 300w, https://educationforasia.org/wp-content/uploads/sites/12/2016/07/education-in-asia-768x328.jpg 768w, https://educationforasia.org/wp-content/uploads/sites/12/2016/07/education-in-asia-1024x438.jpg 1024w, https://educationforasia.org/wp-content/uploads/sites/12/2016/07/education-in-asia-570x244.jpg 570w, https://educationforasia.org/wp-content/uploads/sites/12/sites/12/2016/07/education-in-asia-517x235.jpg 517w`

You can see the latter element.
That is because of this line:
https://github.com/presscustomizr/customizr/blob/v4.0.8/core/front/models/modules/slider/class-model-slider.php#L279

Basically in multisites that _wp_get_attachment_relative_path( $image_src ); returns 
`sites/12/2016/07/`
While we'd expect something like `2016/07/` (that's the case for non multisites).

The solution will be (tested) to use `$image_meta['file']`.

Anyway this gives us also the chance to improve that code to align it to:
https://github.com/WordPress/WordPress/blob/37318fe73f1b32fae1c289738f6d8cc21c99c4f8/wp-includes/media.php#L1044

This issue happens only if there is no tc-slider-small image size available for a given slide

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
…isite

fixes #1247

### CODE REVIEW

**Comment by ghost** [comment]
Yep I should have better checked. thx.
fixed

**Comment by eri-trabiccolo** [comment]
Mmm I think I will re-do this PR. It has some merging issues.

**Comment by ghost** [comment]
no I just merged it. that's fine

**Comment by ghost** [comment]
pushing now

**Comment by ghost** [comment]
needed some intermediate build, that's it

**Comment by eri-trabiccolo** [comment]
Ah ok, thanks :)

File: core/front/models/modules/slider/class-model-slider.php [hunk 1]
```diff
@@ -280,59 +280,103 @@ protected function czr_fn_set_wp_responsive_slide_img_attr() {
...
+                      if( ! isset( $sources[ $image['width'] ] ) ){
+                          $sources[ $image['width'] ] = $source;
+                      }
+                  }
+              }//else
+          }
+      }
 
-    return $sources;
+      return $sources;
```
Comment (ghost) [inline core/front/models/modules/slider/class-model-slider.php:380]: man!
this is not simple, but we need it.
congrats, good piece of code there 👍

**Pull request merged**